android-core/src/main/java/com/mparticle/internal/UploadHandler.java

```diff
@@ -145,8 +151,19 @@ public void handleMessageImpl(Message msg) {
                             }
                         }
                     }
+                    if (!anythingToUpload && (!mAppStateManager.getSession().isActive() || mAppStateManager.isBackgrounded())) {
+                        //If there is anything to upload and is in background or inactive, cancel schedule job until a new message is stored
```

Comment:
Should this say "//If there is **not** anything to upload"? or is the check for `!anythingToUpload` in the if statement incorrect?

Reply:
Good catch